src/res_printer.ml

```diff
@@ -553,18 +553,20 @@ let printConstant ?(templateLiteral = false) c =
   | Pconst_float (s, _) -> Doc.text s
   | Pconst_char c ->
     let str =
-      match c with
-      | '\'' -> "\\'"
-      | '\\' -> "\\\\"
-      | '\n' -> "\\n"
-      | '\t' -> "\\t"
-      | '\r' -> "\\r"
-      | '\b' -> "\\b"
-      | ' ' .. '~' as c ->
-        let s = (Bytes.create [@doesNotRaise]) 1 in
-        Bytes.unsafe_set s 0 c;
-        Bytes.unsafe_to_string s
-      | c -> Res_utf8.encodeCodePoint (Obj.magic c)
+      if c <= 127 then
```

Comment:
why `if c <= 127` needed?

Reply:
I tried to avoid the usage of Char.unsafe_chr. 
Maybe the following code is clearer?
```ml
  let str = 
        match Char.unsafe_chr c with
        | '\'' -> "\\'"
        | '\\' -> "\\\\"
        | '\n' -> "\\n"
        | '\t' -> "\\t"
        | '\r' -> "\\r"
        | '\b' -> "\\b"
        | ' ' .. '~' as c ->
          let s = (Bytes.create [@doesNotRaise]) 1 in
          Bytes.unsafe_set s 0 c;
          Bytes.unsafe_to_string s
        | _ -> Res_utf8.encodeCodePoint c
```